server/app/controllers/applicant/UpsellController.java

```diff
@@ -150,19 +154,35 @@ public CompletionStage<Result> considerRegister(
                   request.flash().get("banner").map(m -> ToastMessage.alert(m));
 
               if (settingsManifest.getNorthStarApplicantUi(request)) {
-                NorthStarApplicantUpsellView.Params params =
-                    NorthStarApplicantUpsellView.Params.builder()
-                        .setRequest(request)
-                        .setProfile(
-                            profile.orElseThrow(
-                                () -> new MissingOptionalException(CiviFormProfile.class)))
-                        .setApplicantPersonalInfo(applicantPersonalInfo.join())
-                        .setProgramTitle(roApplicantProgramService.join().getProgramTitle())
-                        .setApplicationId(applicationId)
-                        .setMessages(messagesApi.preferred(request))
-                        .setApplicantId(applicantId)
-                        .build();
-                return ok(northStarUpsellView.render(params)).as(Http.MimeTypes.HTML);
+                if (isCommonIntake.join()) {
+                  NorthStarApplicantCommonIntakeUpsellView.Params params =
```

Comment:
This looks pretty similar to the upsell view params. Would it make sense to extract those params into a separate class and share them, or are they just different enough that we shouldn't? I'll leave it up to you